src/main/java/org/folio/circulation/domain/policy/RollingRenewalDueDateStrategy.java

```diff
@@ -54,7 +55,10 @@
   @Override
   Result<DateTime> calculateDueDate(Loan loan) {
     if(StringUtils.isBlank(renewFrom)) {
-      return failedValidation(errorForPolicy(RENEW_FROM_UNRECOGNISED_MESSAGE));
+      DateTime dueDate = Optional.ofNullable(loan.getDueDate()).orElse(systemDate);
+      return calculateDueDate(dueDate, loan.getLoanDate()).failed()
```

Comment:
I'm a little confused, if this is turned into a sentence it reads something like:
`If due date calculation fails then report no due date limit schedule, otherwise report that cannot recognise when to renew from`

Can you help me understand what rules this is trying to reflect?

Reply:
I assume that if it is possible to determine if the due date falls in the same range then the error is unknown to us "cannot determine when to renew from" since the "renewFrom" parameter is empty. 
This will keep the previous logic. Otherwise, the renewal date falls outside of date ranges.

Reply:
How does the code know that the only kind of failure during calculating a due date is because it is outside of the fixed schedule due date range?

Reply:
Um, I didn't think about that. Perhaps I should check the due date at the top level, not only in a fixed range. I will try to change the implementation

Reply:
Let me try to rephrase this.

If there is no renew from setting chosen (because the policy states it is not renewable), how can a due date be calculated?

What kind of policy needs to be set up for this two errors to occur together?

Reply:
From my interpretation of scenario 1 in https://issues.folio.org/browse/UIU-965

There needs to be a check for whether the renewal date falls within the fixed due date schedules, irrespective of any attempt to calculate the due date (which I don't think can be done).